src/commands/deployments/connectToGitHub.ts

```diff
@@ -19,13 +20,18 @@ export async function connectToGitHub(context: IActionContext, target?: GenericT
         node = <DeploymentsTreeItem>target.parent;
     }
 
-    await editScmType(node.root.client, node, context, ScmType.GitHub);
-
     if (node instanceof SiteTreeItem) {
+        if (node.deploymentsNode === undefined) {
```

Comment:
Rather than doing all this logic here, I think you can just use `editScmType` and pass whatever node you want.  Note right now that `connectToGitHub` is importing `editScmType` from '`vscode-azureappservice`, but you can use the one that lives in the extension.